File: c/Sum of All Subset XOR Totals.c
#include <stdio.h>

int subsetXORSum(int* nums, int n) {
    int totalSum = 0;
    int subsetCount = 1 << n; 

    for (int i = 0; i < subsetCount; ++i) {
        int currentXOR = 0;
        for (int j = 0; j < n; ++j) {
            if (i & (1 << j)) { 
                currentXOR ^= nums[j]; 
            }
        }
        totalSum += currentXOR;
    }

    return totalSum;
}
// Test cases
int main() {
    // case 1
    int nums1[] = {1, 3};
    int result1 = subsetXORSum(nums1, 2);
    printf("Output for nums = [1, 3]: %d\n", result1); // Expected output: 6
    
    // case 2
    int nums2[] = {5, 1, 6};
    int result2 = subsetXORSum(nums2, 3);
    printf("Output for nums = [5, 1, 6]: %d\n", result2); // Expected output: 28
    
    return 0;
}

File: js/Sum of All Subset XOR Totals.js
var subsetXORSum = function(nums) {
    let total = 0;
    const n = nums.length;
    const count = 1 << n;

    for (let i = 0; i < count; i++) {
        let xor = 0;
        for (let j = 0; j < n; j++) {
            if (i & (1 << j)) xor ^= nums[j];
        }
        total += xor;
    }

    return total;
}
// Test cases
console.log(subsetXORSum([1,3])); // Expected Output: 6
console.log(subsetXORSum([5,1,6])); // Expected Output: 20
console.log(subsetXORSum([3,4,5,6,7,8])); // Expected Output: 480